src/material/datepicker/datepicker-content.html

```diff
@@ -1,6 +1,7 @@
 <div
   cdkTrapFocus
   class="mat-datepicker-content-container"
+  [ngClass]="{'mat-datepicker-custom-header' : datepicker.calendarHeaderComponent}"
```

Comment:
This doesn't need to use `ngClass`, it can be done with a plain `[class.someClass]` binding. Also it seems like the class name doesn't follow the naming convention of other classes on the same element.